src/components/rating/rating.tsx

```diff
@@ -238,27 +254,115 @@ export class Rating
     this.setFocus();
   }
 
-  private updateValue(value: number) {
-    this.value = value;
-    this.calciteRatingChange.emit({ value });
-  }
+  private handleRatingPointerOver = () => {
+    this.isKeyboardInteraction = false;
+  };
 
-  private onKeyboardPressed = (event: KeyboardEvent): void => {
-    if (!this.required && isActivationKey(event.key)) {
-      event.preventDefault();
-      this.updateValue(0);
-    }
+  private handleRatingPointerOut = () => {
+    this.isKeyboardInteraction = true;
+    this.hoverValue = null;
+    this.focusValue = null;
+    this.hasFocus = false;
   };
 
-  private onFocusChange = (selectedRatingValue: number): void => {
+  private handleRatingFocusIn = (): void => {
+    let selectedInput = this.inputFocusRef;
+
+    if (!selectedInput) {
+      const inputs = this.starsMap.map((star) =>
+        this.el.shadowRoot.getElementById(star.id)
+      ) as unknown as HTMLInputElement;
+      let selectedInputIdx = this.value > 0 ? this.value - 1 : 0;
+
+      if (selectedInputIdx == 0) {
+        const average = this.starsMap.filter((i) => i.average);
+
+        if (average.length > 0) {
+          selectedInputIdx = average[average.length - 1].value;
+        }
+      }
+      selectedInput = inputs[selectedInputIdx];
+    }
+
+    const value = Number(selectedInput.value);
+
+    selectedInput.select();
+    this.focusValue = value;
+    this.hoverValue = value;
     this.hasFocus = true;
-    if (!this.required && this.focusValue === selectedRatingValue) {
-      this.updateValue(0);
-    } else {
-      this.focusValue = selectedRatingValue;
+  };
+
+  private handleRatingFocusLeave = (): void => {
+    this.focusValue = null;
+    this.hoverValue = null;
+    this.hasFocus = false;
+  };
+
+  private handleKeyDown = (event: KeyboardEvent) => {
```

Comment:
Can we separate this function into two functions. One for `handleComponentKeyDown` on the Host and one for `handleInputKeyDown` on the input? The host element can only set `isKeyBoardNavigation`?  Then we don't need to check the currentTarget anymore. I'm not sure why the same function is being called for both elements because they do different things.